WordPress/src/main/java/org/wordpress/android/models/Note.java

```diff
@@ -231,9 +231,23 @@ private String getNoticonCharacter() {
         return queryJSON("noticon", "");
     }
 
-    JSONObject getCommentActions() {
+    private JSONObject getCommentActions() {
         if (mActions == null) {
-            mActions = queryJSON("body[last].actions", new JSONObject());
+            // Find comment block that matches the root note comment id
+            long commentId = getCommentId();
+            JSONArray bodyArray = getBody();
+            for (int i = 0; i < bodyArray.length(); i++) {
+                try {
+                    JSONObject bodyItem = bodyArray.getJSONObject(i);
+                    if (bodyItem.has("type") && bodyItem.optString("type").equals("comment")
+                            && commentId == JSONUtil.queryJSON(bodyItem, "meta.ids.comment", 0)) {
+                        mActions = JSONUtil.queryJSON(bodyItem, "actions", new JSONObject());
+                        break;
+                    }
+                } catch (JSONException e) {
+                    break;
+                }
+            }
```

Comment:
I'm not sure if this would make any real performance difference, but would it make sense to detect when `mActions` is null after the for loop and if so set it to any empty `JSONObject`? That way this loop wouldn't be performed again when it isn't necessary.